organicmaps/organicmaps
# [android] Refactor API and Intent processing

- Remove MapTask abstraction in favor of Intents.
- Process intents and API after creation of Drape.
- Restore routing only after creation of Drape (fixes #3441).
- Start location only after creation of Drape (fixes #1128).
- Remove double parsing of `backurl` in URLs.

**Test Plan:**

- Check URLs from http://omaps.app/test.html.
- Check KML/KMZ/GPX files.
- Check "Pick point" from https://github.com/organicmaps/api-android.
- Download Maps -> any downloaded map -> Show on the map.
- Bookmarks -> any list -> click on a blue button the bottom right.
- Bookmarks -> any list -> click on any bookmark.
- Start navigation -> kill the app **via system settings** ->
  start the app again and check that routing planned is started.
